### ISSUE TITLE
Error when doubling the File class

### ISSUE THREAD

**Issue by 8nouk** [bitwes/Gut#147]
The scene is crashing with a `Parser Error: Unknown class "_File"`

Steps to reproduce :
- create a new test
- try doing this : `var file = double(File).new()`
- enjoy the crash

**Comment by bitwes** [comment]
Reproduced the issue.  This has to do with not being able to pass parameters to a constructor when GUT creates the reference instance under the covers.  `File` requires a path to be passed to the constructor.

I think the best approach to address something like this would be to change the `double` method to detect when an instance was passed to it and create the double from that instance instead of making its own.  It would look something like:
```python
func test_with_doubled_file():
  var f = File.new('some_path')
  var DoubledFile = double(f)
  var double_f = DoubledFile.new('some_other_path')
  # <do something>
  assert_called(double_f, 'close')
```
It might be weird that you are making a "class" from an instance.  Any instances created from this class won't have any of the values the source instance did.  If you make a `partial_double` this way you will still have to pass values to the `new` unless `_init` is stubbed `to_do_nothing`.  Also, if you stub `_init` `to_call_super` on a normal double then you will have to pass parameters to `new`.

This feels a bit weird but it should work.  We could add some parameters to the `double` method instead of all this but it already has 2 optional parameters.

**Comment by 8nouk** [comment]
Actually the File constructor doesn't expect parameters.

I temporarily solved the problem by creating a custom class which inherits from File and overrode the methods I wanted to stub directly in it. The resulting class can be doubled so I can test different return values by stubbing it with GUT.

But I also encountered the "cannot deal with constructor parameters" issue. This one is harder to solve.

**Comment by bitwes** [comment]
For some reason `get_class()` on a `File` instance returns `_File` which is then causes the double to blow up when you try to make a double because it starts with `extends _File`.

```
var f = File.new()
print(f.get_class())
```

Not sure how to address this.  There could be more native classes that do this.  I could remove all leading underscores...but maybe there are some that need to be there.  We need a list of all the results of `.get_class()` to make a decision.

**Comment by bitwes** [comment]
After a short discussion on Discord it sounds like stripping off the leading `_` whenever the thing to double is a built in class will work.  At least `File` and all the singletons have the underscore.  I've added this into the next release.

**Comment by Xrayez** [comment]
For the reference, this is related to godotengine/godot#37184, with a suggested fix godotengine/godot#26990.
